compiler/src/dotty/tools/dotc/transform/TypeTestsCasts.scala

```diff
@@ -151,6 +151,7 @@ object TypeTestsCasts {
             // always false test warnings are emitted elsewhere
             X.classSymbol.exists && P.classSymbol.exists &&
               !X.classSymbol.asClass.mayHaveCommonChild(P.classSymbol.asClass)
+            || X.classSymbol.isOneOf(FinalOrSealed)
```

Comment:
`Option` being sealed doesn't mean that `isInstanceOf[Option[Int]]` isn't unchecked.  Except of course that `Unset | Option[Int]` without sealed `Unset` means the scrutinee after case 1 is `Option[Int]`, but I don't think we calculate that in type tests cast, and in a way we can't, because the match tree has been transformed.  There's three places (that I'm aware of) where we look at the match trees: typing, pattern match translation, and space engine analysis.  Perhaps in one of those we can find a way to annotate that the `Option[Int]` test is fully checked.  Even in the case that isn't not the type of the scrutinee, like:

```scala
(v: Unset | Option[Int] | List[Long]) match
  case v: Unset =>
  case v: Option[Int] =>
  case v: List[Long] =>
```